PGD 4 Manual Install DOCS-122 (#5333)
EnterpriseDB/docs: product_docs/docs/pgd/4/admin-manual/installing/07-using-pgd-cli.mdx

---
title: Step 8 - Using PGD CLI
navTitle: Using PGD CLI
deepToC: true
---

## Using PGD CLI

> The PGD CLI command uses a configuration file to work out which hosts to connect to. 

Author: ```suggestion
The PGD CLI client uses a configuration file to work out which hosts to connect to. 
```

There are [options](../../cli/using_cli) that allow you to override this to use alternative configuration files or explicitly point at a server, but by default PGD CLI looks for a configuration file in preset locations.

The connection to the database is authenticated in the same way as other command line utilities, like the psql command, are authenticated. 

> Unlike other commands, PGD CLI doesn't interactively prompt for your password. Therefore, You must pass your password using one of the following methods:

Author: ```suggestion
Unlike other commands, PGD CLI doesn't interactively prompt for your password. Therefore, you must pass your password using one of the following methods:
```

Also consult the [PGD CLI documentation](../../cli/) for details of other configuration options and a full command reference.

## Worked example

### Ensure PGD CLI is installed

In this worked example, we will be configuring and using PGD CLI on host-one, where we've already installed Postgres and PGD.
There is no need to install PGD CLI again. We will be using the `enterprisedb` account as this is already configured for access to Postgres.
If you are not logged in as `enterprisedb` switch to it using:

```
sudo -iu enterprisedb
```

### Create a configuration file

The PGD CLI configuration file is similar to the PGD proxy configuration filer.
It is a YAML file which contains a cluster object. This has two properties:

The name of the PGD cluster's top-level group (as `name`).
An array of endpoints of databases (as `endpoints`).

```
cluster:
  name: pgd
  endpoints:
    - host=host-one dbname=bdrdb port=5444
    - host=host-two dbname=bdrdb port=5444
    - host=host-three dbname=bdrdb port=5444
```

Note that the endpoints in this example specify port=5444.
This is necessary for EDB Postgres Advanced Server instances.

> For EDB Postgres Extended and Community PostgreSQL, this can be omitted.

Author: ```suggestion
For EDB Postgres Extended and Community PostgreSQL, this can be set to `port=5432`.
```
